spikes/input spike/InputHandler.cpp

```diff
@@ -0,0 +1,269 @@
+#include "stdafx.h"
+#include "InputHandler.h"
+using namespace std;
+//this class handles the inputs of the controller to be used outside this class cleanly
+
+//if (inputHandler.getReleasedButton() == InputHandler::BUTTON::A)  <-- use the method to get the key pressed in your class outside this one
+//{}
+InputHandler::InputHandler()
+{
+	wasButtonAPressed = false;
+	wasButtonBPressed = false;
+	wasButtonXPressed = false;
+	wasButtonYPressed = false;
+	wasButtonLeftBumpPressed = false;
+	wasButtonRightBumpPressed = false;
+	wasButtonStartPressed = false;
+	wasButtonSelectPressed = false;
+	lastDpadState = DPadState::CENTER;
+	debug = true;
+}
+void InputHandler::HandleInput()
+{
+	isButtonPressedOnce(PlayerOne);
+}
+void InputHandler::isButtonPressedOnce(int player)
+{
+	//State.reset();
+	bool isButtonAPressed = sf::Joystick::isButtonPressed(player,buttonA);
+	bool isButtonBPressed = sf::Joystick::isButtonPressed(player,buttonB);
+	bool isButtonXPressed = sf::Joystick::isButtonPressed(player,buttonX);
+	bool isButtonYPressed = sf::Joystick::isButtonPressed(player,buttonY);
+	bool isButtonLeftBumpPressed = sf::Joystick::isButtonPressed(player,buttonLeftBumper);
+	bool isButtonRightBumpPressed = sf::Joystick::isButtonPressed(player,buttonRightBumper);
+	bool isButtonSelectPressed = sf::Joystick::isButtonPressed(player,buttonSelect);
+	bool isButtonStartPressed = sf::Joystick::isButtonPressed(player,buttonStart);
+#pragma region Joystic buttom press and release check
+	// doo all the button pressed checks
+	if(isButtonAPressed && !wasButtonAPressed)
+	{	
+		if(debug)
+		{
+			printf("button A : \n");
+		}
+		pressedButton = BUTTON::A;
+	}
+	else if(isButtonBPressed && !wasButtonBPressed)
+	{	
+		if(debug)
+		{
+			printf("button B : \n");
+		}
+		pressedButton = BUTTON::B;
+	}
+	else if(isButtonXPressed && !wasButtonXPressed)
+	{	
+		if(debug)
+		{
+			printf("button X : \n");
+		}
+		pressedButton = BUTTON::X;
+	}
+	else if(isButtonYPressed && !wasButtonYPressed)
```

Comment:
group checks into methods, too messy in one function, also two buttons pressed at the same time may not work with this code